core/trino-main/src/main/java/io/trino/execution/buffer/LazyOutputBuffer.java

```diff
@@ -310,19 +312,31 @@ public void destroy()
     @Override
     public void abort()
     {
+        List<PendingRead> pendingReads = ImmutableList.of();
         OutputBuffer outputBuffer = delegate;
         if (outputBuffer == null) {
             synchronized (this) {
                 if (delegate == null) {
                     // ignore abort if the buffer already in a terminal state.
-                    stateMachine.abort();
+                    if (!stateMachine.abort()) {
+                        return;
+                    }
 
-                    // Do not free readers on fail
```

Comment:
The assumption is that sending `emptyResults(taskInstanceId, 0, completed: false)` is safe, even though the previous comment suggested otherwise, because `TaskResource` will respond similarly when the timeout threshold is reached. Sending these responses instead of letting them time out should propagate task failure information more quickly than allowing the timeout to expire.